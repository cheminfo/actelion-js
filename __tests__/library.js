'use strict';

const core = require('../core');
const minimal = require('../minimal');
const full = require('../full');
const pretty = require('../dist/openchemlib-full.pretty');

describe('Checking for the presence of main APIs', function () {
  const minimalAPI = [
    'Molecule',
    'RingCollection',
    'SDFileParser',
    'SSSearcher',
    'SSSearcherWithIndex',
    'Reaction',
    'Util',
    'version'
  ];

  const coreAPI = [
    'MoleculeProperties',
    'DruglikenessPredictor',
    'DrugScoreCalculator',
    'ToxicityPredictor',
    'ConformerGenerator'
  ];

  const fullAPI = ['StructureView', 'StructureEditor', 'SVGRenderer'];

  // eslint-disable-next-line jest/expect-expect
  it('minimal', function () {
    checkHas(minimal, minimalAPI);
    checkHasNot(minimal, [...coreAPI, ...fullAPI]);
  });

  it('core', function () {
    expect(core).toBe(require('..'));
    checkHas(core, [...minimalAPI, ...coreAPI]);
    checkHasNot(core, fullAPI);
  });

  // eslint-disable-next-line jest/expect-expect
  it('full', function () {
    [full, pretty].forEach((lib) => {
      checkHas(lib, [...minimalAPI, ...coreAPI, ...fullAPI]);
    });
  });
});

function checkHas(obj, properties) {
  expect(Object.keys(obj).sort()).toStrictEqual(properties.sort());
}

function checkHasNot(obj, properties) {
  properties.forEach((prop) => {
    expect(obj).not.toHaveProperty(prop);
  });
}
