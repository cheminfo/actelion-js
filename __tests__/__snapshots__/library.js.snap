// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prototype properties of CanonizerUtil 1`] = `[]`;

exports[`prototype properties of ConformerGenerator 1`] = `
[
  "getConformerCount",
  "getNextConformerAsMolecule",
  "getOCL",
  "getOneConformerAsMolecule",
  "getPotentialConformerCount",
  "getPreviousConformerContribution",
  "initializeConformers",
  "molecules",
]
`;

exports[`prototype properties of DrugScoreCalculator 1`] = `[]`;

exports[`prototype properties of DruglikenessPredictor 1`] = `
[
  "assessDruglikeness",
  "getDetail",
  "getDruglikenessString",
  "getOCL",
]
`;

exports[`prototype properties of ForceFieldMMFF94 1`] = `
[
  "_minimise",
  "getOCL",
  "getTotalEnergy",
  "minimise",
  "size",
]
`;

exports[`prototype properties of Molecule 1`] = `
[
  "addAtom",
  "addBond",
  "addFragment",
  "addImplicitHydrogens",
  "addMissingChirality",
  "addMolecule",
  "addOrChangeAtom",
  "addOrChangeBond",
  "addRing",
  "addRingToAtom",
  "addRingToBond",
  "addSubstituent",
  "calculateTorsion",
  "canonizeCharge",
  "changeAtom",
  "changeAtomCharge",
  "changeBond",
  "clear",
  "convertStereoBondsToSingleBonds",
  "copyAtom",
  "copyBond",
  "copyMolecule",
  "copyMoleculeByAtoms",
  "copyMoleculeByBonds",
  "copyMoleculeProperties",
  "deleteAtom",
  "deleteAtomOrBond",
  "deleteAtoms",
  "deleteBond",
  "deleteBondAndSurrounding",
  "deleteMarkedAtomsAndBonds",
  "deleteMolecule",
  "deleteSelectedAtoms",
  "ensureHelperArrays",
  "findAlleneCenterAtom",
  "findAlleneEndAtom",
  "findAtom",
  "findBINAPChiralityBond",
  "findBond",
  "findRingSystem",
  "getAbsoluteAtomParity",
  "getAbsoluteBondParity",
  "getAllAtoms",
  "getAllBonds",
  "getAllConnAtoms",
  "getAllConnAtomsPlusMetalBonds",
  "getAllHydrogens",
  "getAromaticRingCount",
  "getAtomAbnormalValence",
  "getAtomCIPParity",
  "getAtomCharge",
  "getAtomColor",
  "getAtomCustomLabel",
  "getAtomESRGroup",
  "getAtomESRType",
  "getAtomLabel",
  "getAtomList",
  "getAtomListString",
  "getAtomMapNo",
  "getAtomMass",
  "getAtomParity",
  "getAtomPi",
  "getAtomPreferredStereoBond",
  "getAtomRadical",
  "getAtomRingBondCount",
  "getAtomRingCount",
  "getAtomRingSize",
  "getAtomX",
  "getAtomY",
  "getAtomZ",
  "getAtomicNo",
  "getAtoms",
  "getAverageBondLength",
  "getAverageTopologicalAtomDistance",
  "getBond",
  "getBondAngle",
  "getBondAtom",
  "getBondBridgeMaxSize",
  "getBondBridgeMinSize",
  "getBondCIPParity",
  "getBondESRGroup",
  "getBondESRType",
  "getBondLength",
  "getBondOrder",
  "getBondParity",
  "getBondPreferredStereoBond",
  "getBondQueryFeatures",
  "getBondRingSize",
  "getBondType",
  "getBondTypeSimple",
  "getBonds",
  "getBounds",
  "getCanonizedIDCode",
  "getChiralText",
  "getChirality",
  "getCompactCopy",
  "getConnAtom",
  "getConnAtoms",
  "getConnBond",
  "getConnBondOrder",
  "getDefaultMaxValenceUncharged",
  "getDiastereotopicAtomIDs",
  "getElectronValenceCorrection",
  "getExcludedNeighbourCount",
  "getExplicitHydrogens",
  "getFisherProjectionParity",
  "getFragmentAtoms",
  "getFragmentNumbers",
  "getFragments",
  "getFreeValence",
  "getHandleHydrogenMap",
  "getHelperArrayStatus",
  "getHoseCodes",
  "getIDCode",
  "getIDCodeAndCoordinates",
  "getIDCoordinates",
  "getImplicitHigherValence",
  "getImplicitHydrogens",
  "getIndex",
  "getLowestFreeValence",
  "getMaxAtoms",
  "getMaxBonds",
  "getMaxValence",
  "getMaxValenceUncharged",
  "getMetalBondedConnAtoms",
  "getMolecularFormula",
  "getMoleculeColor",
  "getMolweight",
  "getName",
  "getNonHydrogenNeighbourCount",
  "getNumberOfHydrogens",
  "getOCL",
  "getOccupiedValence",
  "getPath",
  "getPathBonds",
  "getPathLength",
  "getRingSet",
  "getRotatableBondCount",
  "getStereoBond",
  "getStereoCenterCount",
  "getStereoProblem",
  "getSubstituent",
  "getSubstituentSize",
  "getSymmetryRank",
  "getZNeighbour",
  "invalidateHelperArrays",
  "inventCoordinates",
  "is3D",
  "isAlkaliMetal",
  "isAllylicAtom",
  "isAmideTypeBond",
  "isAromaticAtom",
  "isAromaticBond",
  "isAtomConfigurationUnknown",
  "isAtomMarkedForDeletion",
  "isAtomParityPseudo",
  "isAtomStereoCenter",
  "isAutoMappedAtom",
  "isBINAPChiralityBond",
  "isBondBackgroundHilited",
  "isBondBridge",
  "isBondForegroundHilited",
  "isBondMarkedForDeletion",
  "isBondParityPseudo",
  "isBondParityUnknownOrNone",
  "isChalcogene",
  "isDelocalizedBond",
  "isEarthAlkaliMetal",
  "isElectronegative",
  "isElectropositive",
  "isFlatNitrogen",
  "isFragment",
  "isHalogene",
  "isMarkedAtom",
  "isMetalAtom",
  "isNaturalAbundance",
  "isNitrogenFamily",
  "isOrganicAtom",
  "isPseudoRotatableBond",
  "isPurelyOrganic",
  "isRingAtom",
  "isRingBond",
  "isSelectedAtom",
  "isSelectedBond",
  "isSimpleHydrogen",
  "isSmallRingAtom",
  "isSmallRingBond",
  "isStabilizedAtom",
  "isStereoBond",
  "markAtomForDeletion",
  "markBondForDeletion",
  "normalizeAmbiguousBonds",
  "removeAtomColors",
  "removeAtomCustomLabels",
  "removeAtomMarkers",
  "removeAtomSelection",
  "removeBondHiliting",
  "removeExplicitHydrogens",
  "removeQueryFeatures",
  "renumberESRGroups",
  "scaleCoords",
  "setAllAtoms",
  "setAllBonds",
  "setAssignParitiesToNitrogen",
  "setAtomAbnormalValence",
  "setAtomCIPParity",
  "setAtomCharge",
  "setAtomColor",
  "setAtomConfigurationUnknown",
  "setAtomCustomLabel",
  "setAtomESR",
  "setAtomList",
  "setAtomMapNo",
  "setAtomMarker",
  "setAtomMass",
  "setAtomParity",
  "setAtomRadical",
  "setAtomSelection",
  "setAtomX",
  "setAtomY",
  "setAtomZ",
  "setAtomicNo",
  "setBondAtom",
  "setBondBackgroundHiliting",
  "setBondCIPParity",
  "setBondESR",
  "setBondForegroundHiliting",
  "setBondOrder",
  "setBondParity",
  "setBondParityUnknownOrNone",
  "setBondQueryFeature",
  "setBondType",
  "setChirality",
  "setFragment",
  "setHelperValidity",
  "setHydrogenProtection",
  "setMaxAtoms",
  "setMaxBonds",
  "setMoleculeColor",
  "setName",
  "setParitiesValid",
  "setStereoBondFromAtomParity",
  "setStereoBondFromBondParity",
  "setStereoBondsFromParity",
  "setToRacemate",
  "setUnknownParitiesToExplicitlyUnknown",
  "shareSameFragment",
  "stripIsotopInfo",
  "stripSmallFragments",
  "stripStereoInformation",
  "suggestBondType",
  "supportsImplicitHydrogen",
  "swapAtoms",
  "swapBonds",
  "toIsomericSmiles",
  "toMolfile",
  "toMolfileV3",
  "toSVG",
  "toSmiles",
  "translateCoords",
  "validate",
  "zoomAndRotate",
  "zoomAndRotateInit",
]
`;

exports[`prototype properties of MoleculeProperties 1`] = `[]`;

exports[`prototype properties of ReactionEncoder 1`] = `[]`;

exports[`prototype properties of SDFileParser 1`] = `
[
  "getField",
  "getFieldData",
  "getFieldNames",
  "getMolecule",
  "getNextFieldData",
  "getNextMolFile",
  "getOCL",
  "next",
]
`;

exports[`prototype properties of SSSearcher 1`] = `
[
  "getOCL",
  "isFragmentInMolecule",
  "setFragment",
  "setMol",
  "setMolecule",
]
`;

exports[`prototype properties of SSSearcherWithIndex 1`] = `
[
  "createIndex",
  "getOCL",
  "isFragmentInMolecule",
  "setFragment",
  "setMolecule",
]
`;

exports[`prototype properties of SVGRenderer 1`] = `[]`;

exports[`prototype properties of StructureEditor 1`] = `
[
  "getIDCode",
  "getMolFile",
  "getMolFileV3",
  "getMolecule",
  "getOCL",
  "getSmiles",
  "hasFocus",
  "isFragment",
  "onChange",
  "onPasteImage",
  "onPasteString",
  "setAtomHightlightCallback",
  "setBondHightlightCallback",
  "setChangeListenerCallback",
  "setFragment",
  "setIDCode",
  "setMolFile",
  "setSmiles",
]
`;

exports[`prototype properties of StructureView 1`] = `[]`;

exports[`prototype properties of ToxicityPredictor 1`] = `
[
  "assessRisk",
  "getDetail",
  "getOCL",
]
`;

exports[`prototype properties of Transformer 1`] = `
[
  "applyTransformation",
  "setMolecule",
]
`;

exports[`prototype properties of Util 1`] = `[]`;

exports[`static properties of CanonizerUtil 1`] = `
[
  "BACKBONE",
  "NORMAL",
  "NOSTEREO",
  "NOSTEREO_TAUTOMER",
  "TAUTOMER",
  "getIDCode",
]
`;

exports[`static properties of ConformerGenerator 1`] = `
[
  "STRATEGY_ADAPTIVE_RANDOM",
  "STRATEGY_LIKELY_RANDOM",
  "STRATEGY_LIKELY_SYSTEMATIC",
  "STRATEGY_PURE_RANDOM",
]
`;

exports[`static properties of DrugScoreCalculator 1`] = `
[
  "calculate",
]
`;

exports[`static properties of DruglikenessPredictor 1`] = `
[
  "DRUGLIKENESS_UNKNOWN",
]
`;

exports[`static properties of ForceFieldMMFF94 1`] = `
[
  "MMFF94",
  "MMFF94S",
  "MMFF94SPLUS",
]
`;

exports[`static properties of Molecule 1`] = `
[
  "CANONIZER_ASSIGN_PARITIES_TO_TETRAHEDRAL_N",
  "CANONIZER_CONSIDER_DIASTEREOTOPICITY",
  "CANONIZER_CONSIDER_ENANTIOTOPICITY",
  "CANONIZER_CONSIDER_STEREOHETEROTOPICITY",
  "CANONIZER_COORDS_ARE_3D",
  "CANONIZER_CREATE_PSEUDO_STEREO_GROUPS",
  "CANONIZER_CREATE_SYMMETRY_RANK",
  "CANONIZER_DISTINGUISH_RACEMIC_OR_GROUPS",
  "CANONIZER_ENCODE_ATOM_CUSTOM_LABELS",
  "CANONIZER_ENCODE_ATOM_SELECTION",
  "FISCHER_PROJECTION_LIMIT",
  "STEREO_ANGLE_LIMIT",
  "VALIDATION_ERRORS_STEREO",
  "VALIDATION_ERROR_AMBIGUOUS_CONFIGURATION",
  "VALIDATION_ERROR_ESR_CENTER_UNKNOWN",
  "VALIDATION_ERROR_OVER_UNDER_SPECIFIED",
  "cAtomCIPParityNone",
  "cAtomCIPParityProblem",
  "cAtomCIPParityRorM",
  "cAtomCIPParitySorP",
  "cAtomColorBlue",
  "cAtomColorDarkGreen",
  "cAtomColorDarkRed",
  "cAtomColorGreen",
  "cAtomColorMagenta",
  "cAtomColorNone",
  "cAtomColorOrange",
  "cAtomColorRed",
  "cAtomLabel",
  "cAtomParity1",
  "cAtomParity2",
  "cAtomParityIsPseudo",
  "cAtomParityNone",
  "cAtomParityUnknown",
  "cAtomQFAny",
  "cAtomQFAromState",
  "cAtomQFAromStateBits",
  "cAtomQFAromStateShift",
  "cAtomQFAromatic",
  "cAtomQFCharge",
  "cAtomQFChargeBits",
  "cAtomQFChargeShift",
  "cAtomQFExcludeGroup",
  "cAtomQFFlatNitrogen",
  "cAtomQFHydrogen",
  "cAtomQFHydrogenBits",
  "cAtomQFHydrogenShift",
  "cAtomQFMatchStereo",
  "cAtomQFMoreNeighbours",
  "cAtomQFNarrowing",
  "cAtomQFNeighbourBits",
  "cAtomQFNeighbourShift",
  "cAtomQFNeighbours",
  "cAtomQFNoMoreNeighbours",
  "cAtomQFNoOfBits",
  "cAtomQFNot0Hydrogen",
  "cAtomQFNot0Neighbours",
  "cAtomQFNot0PiElectrons",
  "cAtomQFNot1Hydrogen",
  "cAtomQFNot1Neighbour",
  "cAtomQFNot1PiElectron",
  "cAtomQFNot2Hydrogen",
  "cAtomQFNot2Neighbours",
  "cAtomQFNot2PiElectrons",
  "cAtomQFNot2RingBonds",
  "cAtomQFNot3Hydrogen",
  "cAtomQFNot3Neighbours",
  "cAtomQFNot3RingBonds",
  "cAtomQFNot4Neighbours",
  "cAtomQFNot4RingBonds",
  "cAtomQFNotAromatic",
  "cAtomQFNotChain",
  "cAtomQFNotCharge0",
  "cAtomQFNotChargeNeg",
  "cAtomQFNotChargePos",
  "cAtomQFPiElectronBits",
  "cAtomQFPiElectronShift",
  "cAtomQFPiElectrons",
  "cAtomQFRingSize",
  "cAtomQFRingSizeBits",
  "cAtomQFRingSizeShift",
  "cAtomQFRingState",
  "cAtomQFRingStateBits",
  "cAtomQFRingStateShift",
  "cAtomQFRxnParityBits",
  "cAtomQFRxnParityHint",
  "cAtomQFRxnParityInvert",
  "cAtomQFRxnParityRacemize",
  "cAtomQFRxnParityRetain",
  "cAtomQFRxnParityShift",
  "cAtomQFSimpleFeatures",
  "cAtomRadicalState",
  "cAtomRadicalStateD",
  "cAtomRadicalStateNone",
  "cAtomRadicalStateS",
  "cAtomRadicalStateShift",
  "cAtomRadicalStateT",
  "cBondCIPParityEorP",
  "cBondCIPParityNone",
  "cBondCIPParityProblem",
  "cBondCIPParityZorM",
  "cBondParityEor1",
  "cBondParityNone",
  "cBondParityUnknown",
  "cBondParityZor2",
  "cBondQFAllFeatures",
  "cBondQFAromState",
  "cBondQFAromStateBits",
  "cBondQFAromStateShift",
  "cBondQFAromatic",
  "cBondQFBondTypes",
  "cBondQFBondTypesBits",
  "cBondQFBondTypesShift",
  "cBondQFBridge",
  "cBondQFBridgeBits",
  "cBondQFBridgeMin",
  "cBondQFBridgeMinBits",
  "cBondQFBridgeMinShift",
  "cBondQFBridgeShift",
  "cBondQFBridgeSpan",
  "cBondQFBridgeSpanBits",
  "cBondQFBridgeSpanShift",
  "cBondQFDelocalized",
  "cBondQFDouble",
  "cBondQFMatchStereo",
  "cBondQFMetalLigand",
  "cBondQFNarrowing",
  "cBondQFNoOfBits",
  "cBondQFNotAromatic",
  "cBondQFNotRing",
  "cBondQFRing",
  "cBondQFRingSize",
  "cBondQFRingSizeBits",
  "cBondQFRingSizeShift",
  "cBondQFRingState",
  "cBondQFRingStateBits",
  "cBondQFRingStateShift",
  "cBondQFSimpleFeatures",
  "cBondQFSingle",
  "cBondQFTriple",
  "cBondTypeCross",
  "cBondTypeDeleted",
  "cBondTypeDelocalized",
  "cBondTypeDouble",
  "cBondTypeDown",
  "cBondTypeIncreaseOrder",
  "cBondTypeMetalLigand",
  "cBondTypeSingle",
  "cBondTypeTriple",
  "cBondTypeUp",
  "cChiralityDiastereomers",
  "cChiralityEpimers",
  "cChiralityIsomerCountMask",
  "cChiralityKnownEnantiomer",
  "cChiralityMeso",
  "cChiralityNotChiral",
  "cChiralityRacemic",
  "cChiralityUnknown",
  "cChiralityUnknownEnantiomer",
  "cDefaultAtomValence",
  "cESRGroupBits",
  "cESRMaxGroups",
  "cESRTypeAbs",
  "cESRTypeAnd",
  "cESRTypeOr",
  "cHelperBitCIP",
  "cHelperBitIncludeNitrogenParities",
  "cHelperBitNeighbours",
  "cHelperBitParities",
  "cHelperBitRings",
  "cHelperBitSymmetryDiastereotopic",
  "cHelperBitSymmetryEnantiotopic",
  "cHelperBitSymmetrySimple",
  "cHelperBitsStereo",
  "cHelperCIP",
  "cHelperNeighbours",
  "cHelperNone",
  "cHelperParities",
  "cHelperRings",
  "cHelperSymmetryDiastereotopic",
  "cHelperSymmetryEnantiotopic",
  "cHelperSymmetrySimple",
  "cMaxAtomicNo",
  "cMaxConnAtoms",
  "cMoleculeColorDefault",
  "cMoleculeColorNeutral",
  "cRoundedMass",
  "fromIDCode",
  "fromMolfile",
  "fromMolfileWithAtomMap",
  "fromSmiles",
  "getAngle",
  "getAngleDif",
  "getAtomicNoFromLabel",
  "getDefaultAverageBondLength",
  "isAtomicNoElectronegative",
  "isAtomicNoElectropositive",
  "setDefaultAverageBondLength",
]
`;

exports[`static properties of MoleculeProperties 1`] = `[]`;

exports[`static properties of ReactionEncoder 1`] = `
[
  "decode",
  "encode",
]
`;

exports[`static properties of SDFileParser 1`] = `[]`;

exports[`static properties of SSSearcher 1`] = `[]`;

exports[`static properties of SSSearcherWithIndex 1`] = `
[
  "bitCount",
  "getHexStringFromIndex",
  "getIndexFromHexString",
  "getKeyIDCode",
  "getSimilarityAngleCosine",
  "getSimilarityTanimoto",
]
`;

exports[`static properties of SVGRenderer 1`] = `
[
  "renderMolecule",
]
`;

exports[`static properties of StructureEditor 1`] = `
[
  "addPasteHandler",
  "createEditor",
  "createSVGEditor",
  "notify",
]
`;

exports[`static properties of StructureView 1`] = `
[
  "drawMolecule",
  "drawStructure",
  "drawStructureWithText",
  "getIDCode",
  "renderStructure",
  "showStructures",
]
`;

exports[`static properties of ToxicityPredictor 1`] = `
[
  "RISK_HIGH",
  "RISK_LOW",
  "RISK_NAMES",
  "RISK_NO",
  "RISK_UNKNOWN",
  "TYPE_IRRITANT",
  "TYPE_MUTAGENIC",
  "TYPE_REPRODUCTIVE_EFFECTIVE",
  "TYPE_TUMORIGENIC",
]
`;

exports[`static properties of Transformer 1`] = `[]`;

exports[`static properties of Util 1`] = `
[
  "getHoseCodesFromDiastereotopicID",
]
`;
